src/prefect/flows.py

```diff
@@ -2162,6 +2162,11 @@ def my_other_flow(name):
 
     runner = Runner(pause_on_shutdown=pause_on_shutdown, limit=limit, **kwargs)
     for deployment in args:
+        if deployment.work_pool_name:
+            raise ValueError(
```

Comment:
this could be a breaking change in some sense, to avoid that i suggest making this a warning that the work pool argument will be ignored